### DISCUSSION TITLE
feat(env): support installing Node.js latest/rc/nightly

### DISCUSSION THREAD

**Discussion by bompus** [pnpm/pnpm#3857] [Ideas]
Thanks to #3629 , we now have `pnpm env use --global lts` , `pnpm env use --global 16`, `pnpm env use --global 16.11.1`.

I'd like to hear opinions on supporting the following ( which [`n`](https://github.com/tj/n) supports ):

`pnpm env use --global latest` to be inline with `n latest` which installs the version listed as the first array item using `URL 1`. This URL is already being fetched when installing a specific version such as `16.11.1`

`pnpm env use --global rc/v17.0.0-rc.1` to be inline with `n rc/v17.0.0-rc.1` using `URL 2`.

I don't think there is a huge need for nightly, but if so, it'd be the same format as installing an `rc` release - `nightly/v17.0.0-nightly20211012a784258444` using `URL 3`.

In orders of feature priority:

1. `latest` is the lowest barrier to add, because the release json is already being fetched and parsed. `latest` is my most common use case, even above `lts`.
2. `rc` would be a nice-to-have for testing new features and v8 versions before they hit the `release` channel.
3. `nightly` I personally don't have a need for, but it wouldn't be much more to add if `rc` is also supported.

Relevant URLs:
1: https://nodejs.org/download/release/index.json
2: https://nodejs.org/download/rc/index.json
3: https://nodejs.org/download/nightly/index.json

**Comment by zkochan** [comment]
Sounds good

**Reply by zkochan** [reply]
Why do we need to specify the directory? It looks like it is clear from the version what type of Node.js it is. rc always has `-rc.` in the version. nightly always has `-nightly` in the verison

**Reply by bompus** [reply]
There isn't a need for it, but I just figured it would be easier to parse the prefix and then switch off the prefix to determine which /download/.../index.json file to parse.

**Reply by zkochan** [reply]
There are [4 active release directories](https://nodejs.org/download/) (excluding the default one). Each type of directory may be identified from the version's postfix, so it is really enough just to run `pnpm env use -g 16.0.0-rc.1` instead of `pnpm env use -g rc/16.0.0-rc.1`

**Reply by bompus** [reply]
You've convinced me. The style of `pnpm env use -g 17.0.0-rc.3` would work just fine.

**Reply by shadowspawn** [reply]
One advantage of specifying the directory is can then do partial matches like `nightly/v17`.

**Reply by zkochan** [reply]
both will be supported https://github.com/pnpm/pnpm/pull/3892

**Reply by bompus** [reply]
This is awesome. Thanks @zkochan
